core/templates/dev/head/domain/collection/CollectionSkillObjectFactorySpec.js

```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for CollectionNodeObjectFactory.
+ */
+
+describe('Collection skill object factory', function() {
+  var CollectionSkillObjectFactory = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    CollectionSkillObjectFactory = $injector.get(
+      'CollectionSkillObjectFactory');
+  }));
+
+  var _createEmptyCollectionSkill = function(
+      skillId, skillName, questionIds) {
+    return CollectionSkillObjectFactory.create(skillId, {
+      name: skillName,
+      question_ids: questionIds ? questionIds : []
+    });
+  };
+
+  it('should contain initial question ids', function() {
+    var collectionSkill = _createEmptyCollectionSkill(
+      's0', 'skill 1', ['question1', 'question2']);
+    expect(collectionSkill.getId()).toEqual('s0');
+    expect(collectionSkill.getName()).toEqual('skill 1');
+    expect(collectionSkill.getQuestionIds()).toEqual([
+      'question1', 'question2'
+    ]);
+  });
+
+  it('should be able to create a new collection skill by ID and name',
```

Comment:
What about tests for ensuring that changing ``question_ids`` is properly, or not properly reflected in the domain object? We test these sorts of cases with the other collection domain objects to ensure that the internal state of the domain object is not accidentally mutated, but does support Angular binding. Collection node has this for some of its properties. Should question IDs actually be immutable? Whether they should or shouldn't, there should be tests verifying that choice.